### PR TITLE
Improve starter example and add support for Maven Archetypes

### PR DESCRIPTION
**Pull request by m-reza-rahman**
Hi @ivargrimstad and @jeyvison. Here is the initial PR I have been working on. I was going to also issue a PR to update the current GH pages, but quickly realized I should not do that just yet.

I suggest reviewing and merging this PR into the master branch. Then we need to publish the Archetype to Maven Central via the Eclipse Foundation as version 1.0. I can then issue a PR to update the GH pages to reference the Archetype in Maven Central.

Before merging the PR, I suggest tagging the current master branch as v0.1 and creating a branch titled something like "simple", "early" or "archive". The work currently there can then be gradually rolled forward into master and future releases.

After the merge, I suggest doing a 1.0 release from the project. The publish to Maven Central should probably be done via GitHub Actions. Going forward I would also like to set up a nightly build.

### CODE REVIEW

File: src/main/resources/archetype-resources/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM payara/server-full
```
Comment (rdebusscher) [inline src/main/resources/archetype-resources/Dockerfile:1]: This needs a version tag as soon the latest version will be jakarta namespace based.

**Comment by jeyvison** [comment]
Tag and branch with prevous version created. @m-reza-rahman Does this PR close that issue that was beeing used for discusiion? if it does, can you you please use "Closes" word? Thanks :)

File: src/main/resources/archetype-resources/Dockerfile (hunk 1 above)
Comment (m-reza-rahman) [reply src/main/resources/archetype-resources/Dockerfile]: Done.

**Comment by m-reza-rahman** [comment]
Resolves #76

**Comment by m-reza-rahman** [comment]
Thanks, let me start work on GitHub Actions. That will matter for keeping the project stable and doing the releases to Maven Central. I will probably do that next weekend. In the meanwhile, is it possible to work with @ivargrimstad to figure out the mechanics of publishing a release to Maven Central? Perhaps there are existing guidelines and templates we could just use?

**Pull request merged**